src/client/Microsoft.Identity.Client/WwwAuthenticateParameters.cs

```diff
@@ -191,15 +300,129 @@ public static async Task<WwwAuthenticateParameters> CreateFromResourceResponseAs
+                }
+                catch (Exception ex) when (ex is not MsalException)
+                {
+                    throw new MsalClientException(MsalError.UnableToParseAuthenticationHeader, MsalErrorMessage.UnableToParseAuthenticationHeader, ex);
+                }
+            }
+
+            return parameterList;
+        }
+        #endregion Multi Scheme Api
+
+        /// <summary>
+        /// Creates parameters from the WWW-Authenticate string.
+        /// </summary>
+        /// <param name="wwwAuthenticateValue">String contained in a WWW-Authenticate header.</param>
+        /// <param name="scheme">Auth scheme of the result.</param>
+        /// <returns>The parameters requested by the web API.</returns>
+        private static WwwAuthenticateParameters CreateFromWwwAuthenticationHeaderValue(string wwwAuthenticateValue, string scheme)
+        {
+            IDictionary<string, string> parameters = null;
+
+            if (!string.IsNullOrWhiteSpace(wwwAuthenticateValue))
+            {
+                var authValuesSplit = wwwAuthenticateValue.Split(new char[] { ' ' }, 2);
+
+                if (s_knownAuthenticationSchemes.Contains(authValuesSplit[0]))
+                {
+                    parameters = CoreHelpers.SplitWithQuotes(authValuesSplit[1], ',')
+                        .Select(v => AuthenticationHeaderParser.CreateKeyValuePair(v.Trim(), scheme))
+                        .ToDictionary(pair => pair.Key, pair => pair.Value, StringComparer.OrdinalIgnoreCase);
+                }
+                else
+                {
+                    parameters = CoreHelpers.SplitWithQuotes(wwwAuthenticateValue, ',')
+                        .Select(v => AuthenticationHeaderParser.CreateKeyValuePair(v.Trim(), scheme))
+                        .ToDictionary(pair => pair.Key, pair => pair.Value, StringComparer.OrdinalIgnoreCase);
+                }
+            }
+
+            return CreateWwwAuthenticateParameters(parameters, scheme);
+        }
 
-        internal static WwwAuthenticateParameters CreateWwwAuthenticateParameters(IDictionary<string, string> values)
+        internal static WwwAuthenticateParameters CreateWwwAuthenticateParameters(IDictionary<string, string> values, string scheme)
         {
-            WwwAuthenticateParameters wwwAuthenticateParameters = new WwwAuthenticateParameters
+            WwwAuthenticateParameters wwwAuthenticateParameters = new WwwAuthenticateParameters();
+
+            wwwAuthenticateParameters.AuthScheme = scheme;
+
+            if (values == null)
             {
-                RawParameters = values
-            };
+                return wwwAuthenticateParameters;
+            }
+
+            wwwAuthenticateParameters.RawParameters = values;
 
             string value;
+
             if (values.TryGetValue("authorization_uri", out value))
             {
                 wwwAuthenticateParameters.Authority = value.Replace("/oauth2/authorize", string.Empty);
```

Comment:
Should we try to also replace `v2.0` from potential v2 auth endpoints like

`https://login.microsoftonline.com/common/v2.0/oauth2/authorize` ?

Reply:
I'm not sure, is it expected that the `authorization_uri` to come with `v2.0` from the WWW-Auth params. @jmprieur ?